apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -245,6 +245,8 @@ def fabric_l3(self):
         if self._fabric_l3 is None:
             self._fabric_l3 = self._cisco_l3 or hasattr(self._l3_plugin,
                                                         '_apic_driver')
+            # Update the apicapi's property
+            self.apic_manager.apic_config.enable_aci_routing = self._fabric_l3
```

Comment:
This is more a best practices concern. I agree that changing the config is okay right now, but I'd prefer that future code in apicapi (or another consumer of apicapi) did not have to worry about mutability of config state.

I was hoping that fabric_l3 would be driven by the aci_routing_enabled flag instead of the other way around. One could argue that absence of the L3 plugin is indicative of the operator intent to not use ACI for L3, but perhaps it would be more manageable for us if there is a single point that controls that behavior.